src/38query.js

```diff
@@ -8,6 +8,14 @@ function queryfn(query, oldscope, cb, A, B) {
 	query.cb = cb;
 	query.oldscope = oldscope;
 
+	// Clear subquery caches from previous execution
+	// These are used by IN/NOT IN subquery optimization
+	for (let key of Object.keys(query)) {
+		if (key.startsWith('_inCache') || key.startsWith('_notInCache')) {
+			delete query[key];
```

Comment:
Is this really the best way to do it? Why starts with? Cant we just ignore them and reset when we need them in a new query? or even better: a clean up after executing?